tools/tracer/config/config.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2018 Intel Corporation
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include <algorithm>
+#include <ctype.h>
+#include <sstream>
+#include <string.h>
+
+#include "config.h"
+
+#if defined(_WIN32) || defined(_WIN64)
+#include <windows.h>
+const char* get_path_reg()
+    {
+        char* path = new char[128];
+        DWORD size = sizeof(char)*128;
+        DWORD sts = RegGetValue(HKEY_LOCAL_MACHINE, (LPCTSTR)("Software\\Intel\\MediaSDK\\Dispatch\\tracer"), (LPCTSTR)("_conf"), RRF_RT_REG_SZ, (LPDWORD)0, (PVOID)path, (LPDWORD)&size);
+        size_t len = strnlen_s(path, 128);
+
+        if (sts == ERROR_SUCCESS)
+        {
+            for (size_t i = 0; i < len; i++)
+            {
+                    if (path[i] == '\\')
+                        path[i] = '/';
+            }
+
+        }
+        else
+        {
+            GetCurrentDirectory(128, path);
+            len = strnlen_s(path, 128);
+            for (size_t i = 0; i < len; i++)
+            {
+                    if (path[i] == '\\')
+                        path[i] = '/';
+            }
+        }
+        return path;
+    }
+#endif
+
+Config* Config::conf = NULL;
+
+Config::Config()
+{
+    const char* home =
```

Comment:
who will delete dynamic memory returned from get_path_reg() (on windows)?

Reply:
Windows support out of scope for this branch. But code in question contains memory leak. Not that critical, considering debugging nature of the tracer, but still should be fixed.